framework/QuartzRenderer/QuartzRenderer.swift

```diff
@@ -0,0 +1,599 @@
+                                       blue: CGFloat(strokeColor.b),
+                                       alpha: CGFloat(strokeColor.a)))
+        context.setLineWidth(CGFloat(thickness))
+        context.addPath(line)
+        if(isDashed) {
+            let dashes: [ CGFloat ] = [ CGFloat(thickness + 1), CGFloat(thickness + 1) ]
+            context.setLineDash(phase: 1, lengths: dashes)
+        }
+        context.strokePath()
+        context.setLineDash(phase: 1, lengths: [])
+    }
+
+    public func drawPlotLines(points p: [Point],
+                              strokeWidth thickness: Float,
+                              strokeColor: Color,
+                              isDashed: Bool) {
+        for i in 0..<p.count-1 {
+            drawLine(startPoint: p[i],
+                     endPoint: p[i+1],
+                     strokeWidth: thickness,
+                     strokeColor: strokeColor,
+                     isDashed: isDashed,
+                     isOriginShifted: true)
+        }
+    }
+
+    public func drawText(text s: String,
+                         location p: Point,
+                         textSize size: Float,
+                         strokeWidth thickness: Float,
+                         angle: Float,
+                         isOriginShifted: Bool){
+        var x1 = p.x + xOffset
+        var y1 = p.y + yOffset
+        if (isOriginShifted) {
+            x1 = x1 + 0.1*plotDimensions.subWidth
+            y1 = y1 + 0.1*plotDimensions.subHeight
+        }
+        let font = NSFont.systemFont(ofSize: CGFloat(size))
+//        let transform = CGAffineTransform(rotationAngle: CGFloat(angle))
+//        context.textMatrix = transform
+        let attr:CFDictionary = [NSAttributedString.Key.font:font,NSAttributedString.Key.foregroundColor:NSColor.black] as CFDictionary
+        let cfstring:CFString = s as NSString
+        let text = CFAttributedStringCreate(nil, cfstring, attr)
+        let line = CTLineCreateWithAttributedString(text!)
+        context.setLineWidth(1)
+        context.setTextDrawingMode(.fill)
+        context.textPosition = CGPoint(x: Double(x1), y: Double(y1))
+        CTLineDraw(line, context)
+//        context.textMatrix = CGAffineTransform(rotationAngle: 0)
+    }
+
+    public func getTextWidth(text: String,
+                             textSize s: Float) -> Float {
+        let font = NSFont.systemFont(ofSize: CGFloat(s))
+        context.setFont(CTFontCopyGraphicsFont(font, nil))
+        let string = NSAttributedString(string: "\(text)", attributes: [NSAttributedString.Key.font: font])
+        let size = string.size()
+        return Float(size.width)
+    }
+
+    public func drawOutput(fileName name: String) {
+        if !name.isEmpty {
+            let fileName = name + ".png"
+            let destinationURL = URL(fileURLWithPath: fileName)
+            let image = NSImage(cgImage: context.makeImage()!, size: NSZeroSize)
```

Comment:
You can do an `#if os(macOS)` here to switch between the two image types. If you made the extensions match between them (below), the rest of the code could remain outside of the platform conditional.